model-optimizer/extensions/ops/non_max_suppression.py

```diff
@@ -96,10 +96,9 @@ def infer(node: Node):
         node.out_port(0).data.set_shape(shape_array([max_number_of_boxes, 3]))
 
         if opset == 'opset5':
-            node.out_port(0).data.set_shape(shape_array([dynamic_dimension_value, 3]))
             num_of_outputs = len([port for port in node.out_ports().values() if not port.disconnected()])
             if num_of_outputs >= 2 and node.has_port('out', 1):
-                node.out_port(1).data.set_shape(shape_array([dynamic_dimension_value, 3]))
+                node.out_port(1).data.set_shape(shape_array([max_number_of_boxes, 3]))
```

Comment:
According to specification plugins that don't support dynamism produce shape:
[min(num_boxes, max_output_boxes_per_class) * num_batches * num_classes, 3]
Is this code correct when num_boxes < max_output_boxes_per_class?

And also, is it correct to set exact shape here? NMS output is dynamic in general and depends on NMS thresholds. According to specification exact shape is set only for plugins that do not support dynamism. But MO should support dynamism, so I don't understand why this change is needed.

Reply:
But in nGraph implementation we calculate and set exact value if some values are defined: https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/non_max_suppression.cpp#L813
But in current MO we set dynamic dimension in all cases.